connector_search_engine/models/se_binding.py

```diff
@@ -103,14 +105,63 @@ def unlink(self):
             self.env["se.binding.todelete"].sudo().create(todelete_vals_list)
         return super().unlink()
 
-    def jobify_recompute_json(self, force_export=False):
-        description = _("Recompute %s json and check if need update" % self._name)
+    def _get_binding_to_process(self, bindings, batch_size):
+        return bindings[0:batch_size], bindings[batch_size:]  # noqa: E203
+
+    def jobify_recompute_json(self, force_export=False, batch_size=500):
         # The job creation with tracking is very costly. So disable it.
-        for record in self.with_context(tracking_disable=True):
-            record.with_delay(description=description).recompute_json(
+        bindings = self.with_context(tracking_disable=True)
+        while bindings:
+            processing, bindings = self._get_binding_to_process(bindings, batch_size)
+
+            # We check if we are currently handling an exception in order
+            # to change the job description.
+            (current_exception, current_exception_value, __) = sys.exc_info()
```

Comment:
line 119 to 141 can be moved to `_jobify_get_job_description` or alike to make the whole thing more readable